### ISSUE TITLE
unused_import lint triggered when the item is used in documentation

### ISSUE THREAD

**Issue by scrabsha** [rust-lang/rust#106987]
### Code

```Rust
// The following import is not used in the code. However, it is used in the
// documentation for `_Useless`.
use std::borrow::Cow;

/// This is a doc comment
///
/// The [`Cow`] says moo.
struct _Useless;
```

### Current output

```Shell
Compiling playground v0.0.1 (/playground)
warning: unused import: `std::borrow::Cow`
 --> src/lib.rs:3:5
  |
3 | use std::borrow::Cow;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: `playground` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.59s
```

### Desired output

```Shell
Compiling playground v0.0.1 (/playground)
    Finished dev [unoptimized + debuginfo] target(s) in 0.48s
```

### Rationale and extra context

In this example, removing this import breaks the documentation, as it uses intra-doc links. As a result, `rustc` should not warn against unused import.

Perhaps `rustc` should parse the documentation in order to find usages of additional imported items.

**Comment by fmease** [comment]
Regardless of whether this is a bug or not, as a workaround you can slap `#[cfg(doc)]` onto `use std::borrow::Cow`.

**Comment by petrochenkov** [comment]
Closing as wontfix, rustc is not obliged to inspect documentation or have any tools for parsing markdown in general, so either full paths in doc links, or `cfg`s or `allow` should be used in cases like this.
